### DISCUSSION TITLE
large number of deprecated packages installed via `npm install` -- including multiple CVEs

### DISCUSSION THREAD

**Discussion by rossgrady** [rancher-sandbox/rancher-desktop#4123] [General]
Do y'all have a published policy about how you manage your NPM dependencies, including an upgrade cadence, etc? 

When I clone the `main` branch of the repo & do an `npm install` I see a lot of deprecation warnings. And when I run a Mend scan against the resulting directory with its node modules populated, there are quite a few `high` CVEs in transitive dependencies.

**Comment by rossgrady** [comment]
So, for example, there is a CVE for `postcss` version 5.2.18 -- when I do `npm list postcss@5.2.18` this is the tree it returns:
```
└─┬ @rancher/shell@0.1.3
  └─┬ require-extension-hooks-vue@3.0.0
    └─┬ postcss-modules-sync@1.0.0
      └── postcss@5.2.18 
```

**Reply by jandubois** [reply]
There is no published policy, but in general we run `dependabot` and update modules whenever possible, especially if they include CVE fixes.

In this particular instance the fixes for `postcss` are only available in `postcss@7.0.36` or `postcss@8.2.13`. Unfortunately these are incompatible with the current version of Vue, and upgrading Vue will require some major refactoring. Since Rancher Desktop is using components from other Rancher projects, this needs to be coordinated across those teams.

On the mitigating side, `postcss` is only used during the build process and not at runtime, so the vulnerability should not be a concern for the end-user.

Even in other cases, vulnerabilities that would be a concern in a web application may not be exploitable against a local Electron app, which is running with regular user privileges and not accessible from outside the machine.

**Reply by rossgrady** [reply]
Thanks for the followup & the detailed explanation! This was very helpful. (also, side note, I definitely feel the pain w/r/t Vue & its dependencies!)
